packages/amplify-provider-awscloudformation/src/graphql-transformer/transform-config.ts

```diff
@@ -0,0 +1,196 @@
+import * as path from 'path';
+import { TransformConfig, TransformerProjectConfig } from '@aws-amplify/graphql-transformer-core';
+import fs from 'fs-extra';
+
+export interface ProjectOptions {
+  projectDirectory?: string;
+  transformersFactory: Function;
+  transformersFactoryArgs: object[];
+  currentCloudBackendDirectory: string;
+  rootStackFileName?: string;
+  dryRun?: boolean;
+  disableFunctionOverrides?: boolean;
+  disablePipelineFunctionOverrides?: boolean;
+  disableResolverOverrides?: boolean;
+  buildParameters?: Object;
+  minify?: boolean;
+}
+
+
+export const TRANSFORM_CONFIG_FILE_NAME = `transform.conf.json`;
+
+/**
+ * try to load transformer config from specified projectDir
+ * if it does not exist then we return a blank object
+ *  */
+
+export async function loadConfig(projectDir: string): Promise<TransformConfig> {
+  // Initialize the config always with the latest version, other members are optional for now.
+  let config: TransformConfig = {
+    schema: ''
+  };
+  try {
+    const configPath = path.join(projectDir, TRANSFORM_CONFIG_FILE_NAME);
+    const configExists = fs.existsSync(configPath);
+    if (configExists) {
+      const configStr = await fs.readFile(configPath, 'utf-8');
+      config = JSON.parse(configStr);
+    }
+    return config as TransformConfig;
+  } catch (err) {
+    return config;
+  }
+}
+
+export async function writeConfig(projectDir: string, config: TransformConfig): Promise<TransformConfig> {
+  const configFilePath = path.join(projectDir, TRANSFORM_CONFIG_FILE_NAME);
+  await fs.writeFile(configFilePath, JSON.stringify(config, null, 4));
+  return config;
+}
+
+/**
+ * Given an absolute path to an amplify project directory, load the
+ * user defined configuration.
+ */
+
+
+export async function loadProject(projectDirectory: string, opts?: ProjectOptions): Promise<TransformerProjectConfig> {
+  // Schema
+  const schema = await readSchema(projectDirectory);
+
+  // Load functions
+  const functions = {};
+  if (!(opts && opts.disableFunctionOverrides === true)) {
+    const functionDirectory = path.join(projectDirectory, 'functions');
+    const functionDirectoryExists = fs.existsSync(functionDirectory);
+    if (functionDirectoryExists) {
+      const functionFiles = await fs.readdir(functionDirectory);
+      for (const functionFile of functionFiles) {
+        if (functionFile.indexOf('.') === 0) {
+          continue;
+        }
+        const functionFilePath = path.join(functionDirectory, functionFile);
+        functions[functionFile] = functionFilePath;
```

Comment:
this logic looks like it's repeated below a couple times. Can it be made into a helper function?
At the very least, I think the indexOf('.') check could be pulled out and used like:
```js
functionFiles.filter(fileNotHidden).forEach(...)

fileNotHidden = (file: string) => file.indexOf('.') !== 0
```